### ISSUE TITLE
Imperative mood on auxiliaries

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/UD_English-EWT#326]
EWT marks `Mood=Imp` on "do" auxiliaries when used in an imperative clause. Is this correct, and if so, what should be the features of the main verb—`Mood=Imp|VerbForm=Fin` or `VerbForm=Inf`?

EWT uses `Mood=Imp|VerbForm=Fin` on the main verb in addition to the AUX: http://universal.grew.fr/?custom=62a6a0f450257

With "do" plus a passive auxiliary, both are marked imperative: http://universal.grew.fr/?custom=62a6a35dc5cd7

By contrast, GUM uses `Mood=Imp` only on VERBs, never AUXes: http://universal.grew.fr/?custom=62a6a158e7d34 This seems like a problem at minimum because it excludes imperative uses of "be". For example: http://universal.grew.fr/?custom=62a6a4e7c7c71

**Comment by amir-zeldes** [comment]
Agreed, imperative 'be' should receive `Mood=Imp`. There is only one case in GUM ATM, but that's because the morph script quashes the other cases, will fix.

**Comment by amir-zeldes** [comment]
About the questions above:
  * Imperative aux do should be Imp IMO (why not?)
  * In general I think only the aux should have a Mood if the lexical verb is an infinitive
  * Similarly in the passive double aux case, only "do" should be Imp (the other aux is an infinitive, so it does not have a mood in the morphological sense). 

This is not only historically and traditionally correct (we put Imp only on the verbs which would carry Imp mood in the IE sense), but also makes sure that counting Imp in the corpus results in the correct number.

**Comment by nschneid** [comment]
> Agreed, imperative 'be' should receive `Mood=Imp`. There is only one case in GUM ATM, but that's because the morph script quashes the other cases, will fix.

(just fixed one of my GUM links above)

**Comment by nschneid** [comment]
The above policy seems reasonable if we are marking `Mood=Imp` on verbs. Without thinking about other languages it is hard to intuit which verb/aux should receive that designation, because in English the is no distinction between the imperative form and the plain form (i.e. `VB` in PTB). So one could make the argument that in English, imperative is properly a clause type, not a morphological feature (cf. UniversalDependencies/docs#877). But the "infinitive" terminology of `VerbForm=Inf` would be counterintuitive here because we say imperative clauses are finite, and I don't know how else to indicate the plain form of the verb.

**Comment by amir-zeldes** [comment]
A lot of English morphology is stricty speaking 'hypothetical' and practically speaking simply a continuation of Indo-European linguistic terminology and tradition. But there are advantages to keeping this tradition: it's probably the least confusing to non-UD linguists new to the framework, and it has the benefit of being comparable to related languages, where the IE system is more overtly in evidence. Historically speaking, only the subjunctive and imperative 'be', as wel as the inflected modals, are overt reflexes of the IE mood system, but systemically I think it makes sense to use `Imp` for lexical verbs where 'be' could stand if the verb were 'be' (and historically, these used to be marked in older English). Infinitives do not have mood in the IE system in any language can think of (or incidentally in its nearest neighbor, Afro-Asiatic, either), so diverging from that would probably be unexpected.
